hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -326,6 +345,70 @@ public void createDbStore(MutableVolumeSet dbVolumeSet) throws IOException {
     }
   }
 
+  /**
+   * Ensure that volume is initialized properly with
+   * cleanup path. Should disk be re-inserted into
+   * cluster, cleanup path should already be on
+   * disk. This method syncs the HddsVolume
+   * with the path on disk.
+   * 
+   * @param id clusterId or scmId
+   */
+  public void checkTmpDirPaths(String id) {
```

Comment:
@errose28 Tmp directories are only created when needed and not along with volume initialization. Without this method setting the paths everything breaks.

Reply:
@errose28 The tmp is created when creating a working dir. There are cases were this step is missed because working dir already exists but the paths used for the tmp dir are null and we need to set them again. Check a workflow I launched where we are skipping this method.

https://github.com/xBis7/ozone/actions/runs/3874176967/jobs/6605289826

Reply:
@xBis7 , we can remove the need for the `checkTmpDir` methods in the `HddsVolume` with a few changes that cleanup the code and also resolves this issue.  
1.) remove `HddsVolume.checkTmpPaths` 
2.) remove `checkCleanupDirs `method and calls in `StorageVolumeUtil`
3.) in `HddsVolume.createTmpPath` get the `id` from a call to new` HddsVolume` method,
`private String getIdDir() `that returns versions id : `return VersionedDatanodeFeatures.ScmHA.chooseContainerPathID(this, getClusterID() == null ? "" : getClusterID());`
4.) In `keyValueContainerUtil.getLeftovers` and `keyValueContainerUtil.moveToTmpDirectory` call `createDeleteServiceDir `instead of `checkTmpPaths`.  The `createDeleteServiceDir` will check whether the path exists in hddsvolume intstance or on disk as we want it to.

With the above changes, the code is simplified in the datanode start `VersionEndpointTask`, we no longer need the checkTmpPaths, it takes into account the volume path layout for different versions (scmid, clusterid) and it seems to pass all the unit tests.  I'll supply you with the diff.  Note the unit test failures in the current CI workflow.

Reply:
@neils-dev This approach is simpler and seems to work as expected. Thanks!